src/EventGrid/EventGrid/EventSubscription/NewAzureEventGridSubscription.cs

```diff
@@ -783,6 +783,13 @@ public override void ExecuteCmdlet()
                     WriteWarning(EventGridConstants.EventSubscriptionHandshakeValidationMessage);
                 }
 
+                if (this.IncludedEventType != null && this.IncludedEventType.Length == 1 && string.Equals(this.IncludedEventType[0], "All", StringComparison.OrdinalIgnoreCase))
```

Comment:
@ahamad-MS out of curiosity, is there a closed set of values that the user can provide to the `-IncludedEventType` parameter? If so, we should use a generic [`PSArgumentCompleter`](https://github.com/cormacpayne/azure-powershell/blob/docs-update/documentation/development-docs/design-guidelines/parameter-best-practices.md#generic-argument-completer) to specify these values and allow users to tab over them.

Reply:
There is no fixed list here. We have different lists for different topic types and even these lists are changing continuously. So we cannot use PSArgumentCompleter here efficiently.